lib/aggregate.js

```diff
@@ -1050,7 +1050,7 @@ Aggregate.prototype.exec = async function exec() {
   let result;
   try {
     const cursor = collection.aggregate(this._pipeline, options);
-    result = await cursor.toArray();
+    result = await (await cursor).toArray();
```

Comment:
Actually very unsexy. But it is what it is.

Reply:
I'd prefer `await cursorPromise.then(cursor => cursor.toArray())`